tests/jdom/button.spec.ts

```diff
@@ -0,0 +1,30 @@
+// Test the ButtonServer, intended as a test for the unit test framework.
+// Yes, we're meta-testing!
+
+import { suite, test } from "mocha"
+import { ButtonEvent } from "../../src/jdom/constants";
+import {withBus, JDBusTestUtil} from "./tester";
+import { assert } from "../../src/jdom/utils";
+import ButtonServer from "../../src/servers/buttonserver";
+
+
+suite('button server', () => {
+    test('fires edge events after changing state', async function() {
+        const buttonServer = new ButtonServer("button", false)
+
+        await withBus([
+            {server: buttonServer},
+        ], async (bus, serviceMap) => {
+            const busTest = new JDBusTestUtil(bus)  // TODO needs better name, also boilerplate?
+            const buttonService = serviceMap.get(buttonServer)  // TODO boilerplate, think about how to eliminate
+
+            buttonServer.down()
+            assert((await busTest.nextEventWithin(buttonService, {within: 100})).code == 
```

Comment:
it gets convoluted with all the parenthesis and await. it would be better to break this into multiple statements

Reply:
Are you looking for something more of
```javascript
event = await busTest.nextEventWithin(buttonService, {within: 100})
assert(event.code == ButtonEvent.Down)?
```
?

I feel like that this is all one conceptual object, though it's also a long line with a lot of stuff in it...